central/sensor/service/connection/connection_impl.go

```diff
@@ -500,6 +502,56 @@ func (c *sensorConnection) getClusterConfigMsg(ctx context.Context) (*central.Ms
 	}, nil
 }
 
+func (c *sensorConnection) getScanConfigurationMsg(ctx context.Context) (*central.MsgToSensor, error) {
+	_, exists, err := c.clusterMgr.GetCluster(ctx, c.clusterID)
+	if err != nil {
+		return nil, err
+	}
+	if !exists {
+		return nil, errors.Errorf("could not pull config for cluster %q because it does not exist", c.clusterID)
+	}
+
+	q := search.NewQueryBuilder().AddExactMatches(search.ClusterID, c.clusterID).ProtoQuery()
+	scanSettingDS := compScanSetting.Singleton()
+	scanConfigs, err := scanSettingDS.GetScanConfigurations(ctx, q)
+	if err != nil {
+		return nil, err
+	}
+
+	var reformatedConfigs []*central.ApplyComplianceScanConfigRequest
+	for _, scanConfig := range scanConfigs {
+		var profiles []string
+		for _, profile := range scanConfig.Profiles {
+			profiles = append(profiles, profile.ProfileName)
+		}
+		scanConfigRequest := central.ApplyComplianceScanConfigRequest{
+			ScanRequest: &central.ApplyComplianceScanConfigRequest_ScheduledScan_{
```

Comment:
This is the incorrect message. You can check the expected message in sensor's side or the integration tests

Reply:
Done, thanks